src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/actions/manual/ManualTriggerActionApprovalTest.java

```diff
@@ -0,0 +1,225 @@
+        System.setProperty(PluginImpl.TEST_SSH_KEYFILE_LOCATION_PROPERTY, sshKey.getPrivateKey().getAbsolutePath());
+        server = new SshdServerMock();
+        sshd = SshdServerMock.startServer(port, server);
+        server.returnCommandFor("gerrit ls-projects", SshdServerMock.EofCommandMock.class);
+        server.returnCommandFor("gerrit version", SshdServerMock.EofCommandMock.class);
+        server.returnCommandFor("gerrit query --format=JSON --current-patch-set \"status:open\"",
+            SshdServerMock.SendQueryLastPatchSet.class);
+        server.returnCommandFor("gerrit query --format=JSON --patch-sets --current-patch-set \"status:open\"",
+            SshdServerMock.SendQueryAllPatchSets.class);
+    }
+
+    /**
+     * Runs after test method.
+     *
+     * @throws Exception throw if so.
+     */
+    @After
+    public void tearDown() throws Exception {
+        sshd.stop(true);
+        sshd = null;
+    }
+
+    /**
+     * Tests {@link ManualTriggerAction#getApprovals(net.sf.json.JSONObject, int)}.
+     * With an last patchset
+     * @throws Exception if so.
+     */
+    @Test
+    public void testDoGerritSearchLastPatchSet() throws Exception {
+        GerritServer gerritServer = new GerritServer(gerritServerName);
+        PluginImpl.getInstance().addServer(gerritServer);
+        gerritServer.start();
+
+        Config config = (Config)gerritServer.getConfig();
+        config.setGerritAuthKeyFile(sshKey.getPrivateKey());
+        config.setGerritHostName("localhost");
+        config.setGerritSshPort(port);
+        gerritServer.startConnection();
+
+        FreeStyleProject project = DuplicatesUtil.createGerritTriggeredJob(j, projectName, gerritServerName);
+
+        final ManualTriggerAction action = new ManualTriggerAction();
+        HttpSession session = mock(HttpSession.class);
+
+        StaplerRequest request = mock(StaplerRequest.class);
+        StaplerResponse response = mock(StaplerResponse.class);
+
+        when(request.getSession(true)).thenReturn(session);
+
+        doAnswer(new Answer() {
+            /**
+             * @see org.mockito.stubbing.Answer#answer(org.mockito.invocation.InvocationOnMock)
+             */
+            @Override
+            public Object answer(InvocationOnMock aInvocation) throws Throwable {
+                List<JSONObject> json = (List<JSONObject>)aInvocation.getArguments()[1];
+
+                ManualTriggerAction.HighLow highLow = action.getCodeReview(json.get(0), 2);
+                assertEquals(2, highLow.getHigh());
+                assertEquals(-1, highLow.getLow());
+
+                highLow = action.getVerified(json.get(0), 2);
+                assertEquals(2, highLow.getHigh());
+                assertEquals(-1, highLow.getLow());
+
+                return null;
+            }
+
+        }).when(session).setAttribute(eq("result"), any());
+
+        action.doGerritSearch("status:open", gerritServerName, false, request, response);
```

Comment:
It does test when session.setAttribute is invoked. This is only one result of action.doGerritSearch.
There are assertEquals.